Scripts/Services/LanguageTranslator/V3/LanguageTranslatorService.cs

```diff
@@ -646,28 +726,121 @@ public bool ListModels(Callback<TranslationModels> callback, string source = nul
+        {
+            DetailedResponse<DocumentList> response = new DetailedResponse<DocumentList>();
+            foreach (KeyValuePair<string, string> kvp in resp.Headers)
+            {
+                response.Headers.Add(kvp.Key, kvp.Value);
+            }
+            response.StatusCode = resp.HttpResponseCode;
+
+            try
+            {
+                string json = Encoding.UTF8.GetString(resp.Data);
+                response.Result = JsonConvert.DeserializeObject<DocumentList>(json);
+                response.Response = json;
+            }
+            catch (Exception e)
+            {
+                Log.Error("LanguageTranslatorService.OnListDocumentsResponse()", "Exception: {0}", e.ToString());
+                resp.Success = false;
+            }
+
+            if (((RequestObject<DocumentList>)req).Callback != null)
+                ((RequestObject<DocumentList>)req).Callback(response, resp.Error);
+        }
+        /// <summary>
+        /// Translate document.
+        ///
+        /// Submit a document for translation. You can submit the document contents in the `file` parameter, or you can
+        /// reference a previously submitted document by document ID.
+        /// </summary>
+        /// <param name="callback">The callback function that is invoked when the operation completes.</param>
+        /// <param name="file">The source file to translate.
+        ///
+        /// [Supported file
+        /// types](https://cloud.ibm.com/docs/services/language-translator?topic=language-translator-document-translator-tutorial#supported-file-formats)
+        ///
+        /// Maximum file size: **20 MB**.</param>
+        /// <param name="fileContentType">The content type of file. (optional)</param>
+        /// <param name="modelId">The model to use for translation. `model_id` or both `source` and `target` are
+        /// required. (optional)</param>
+        /// <param name="source">Language code that specifies the language of the source document. (optional)</param>
+        /// <param name="target">Language code that specifies the target language for translation. (optional)</param>
+        /// <param name="documentId">To use a previously submitted document as the source for a new translation, enter
+        /// the `document_id` of the document. (optional)</param>
+        /// <returns><see cref="DocumentStatus" />DocumentStatus</returns>
+        public bool TranslateDocument(Callback<DocumentStatus> callback, System.IO.MemoryStream file, string filename,  string fileContentType = null, string modelId = null, string source = null, string target = null, string documentId = null)
+        {
+            if (callback == null)
+                throw new ArgumentNullException("`callback` is required for `TranslateDocument`");
+            if (file == null)
+                throw new ArgumentNullException("`file` is required for `TranslateDocument`");
```

Comment:
Can you add a check for `filename` here? I think we need to add support for adding the check in the generator.
```cs
if (string.IsNullOrEmpty(filename))
    throw new ArgumentNullException("`filename` is required for `TranslateDocument`");
```